### PR TITLE
Async call batcher mutation /4713

### PR DESCRIPTION
**Pull request by pchauhan-qlogic**
issue 4713

### CODE REVIEW

File: bigtable/google/cloud/bigtable/table.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 """User-friendly container for Google Cloud Bigtable Table."""
-
+import threading
 
 from grpc import StatusCode
```
Comment (mf2199) [inline bigtable/google/cloud/bigtable/table.py:19]: Is the blank line necessary here?

File: bigtable/google/cloud/bigtable/table.py [hunk 2]
```diff
@@ -50,6 +50,7 @@
 #  google.bigtable.v2#google.bigtable.v2.MutateRowRequest)
 _MAX_BULK_MUTATIONS = 100000
 VIEW_NAME_ONLY = enums.Table.View.NAME_ONLY
+_MAX_THREAD_LIMIT = 10
```
Comment (mf2199) [inline bigtable/google/cloud/bigtable/table.py:51]: Why is this 10 and not, say, 5 ro 20?

File: bigtable/google/cloud/bigtable/table.py [hunk 3]
```diff
@@ -511,10 +512,13 @@ def mutate_rows(self, rows, retry=DEFAULT_RETRY):
                   corresponding to success or failure of each row mutation
                   sent. These will be in the same order as the `rows`.
         """
+
```
Comment (mf2199) [inline bigtable/google/cloud/bigtable/table.py]: This blank line can be removed.

File: bigtable/google/cloud/bigtable/table.py [hunk 4]
```diff
@@ -671,13 +675,20 @@ class _RetryableMutateRowsWorker(object):
     )
     # pylint: enable=unsubscriptable-object
 
-    def __init__(self, client, table_name, rows, app_profile_id=None, timeout=None):
+    def __init__(self, client, table_name, batches, app_profile_id=None, timeout=None):
+
```
Comment (mf2199) [inline bigtable/google/cloud/bigtable/table.py]: You could add a comment here. Follow the standard commenting rules. See another GCP function for an example.

File: bigtable/google/cloud/bigtable/table.py [hunk 5]
```diff
@@ -690,26 +701,46 @@ def __call__(self, retry=DEFAULT_RETRY):
...
+            thread = threading.Thread(target=self.thread_func, args=(retry,))
+            thread.start()
+            thread_list.append(thread)
+        for thread in thread_list:
+            thread.join()
 
         return self.responses_statuses
 
+    def thread_func(self, retry=DEFAULT_RETRY):
+
```
Comment (mf2199) [inline bigtable/google/cloud/bigtable/table.py:728]: Please add a description comment here. See another GCP function for a formatting example and follow the same format.

**Review by mf2199 — CHANGES_REQUESTED** [review]

File: bigtable/tests/unit/test_table.py [hunk 6]
```diff
@@ -1152,9 +1161,9 @@ def test_callable_no_retry_strategy(self):
         with mock.patch("google.cloud.bigtable.table.wrap_method") as patched:
             patched.return_value = mock.Mock(return_value=[response])
 
-            worker = self._make_worker(client, table.name, [row_1, row_2, row_3])
+            worker = self._make_worker(client, table.name, [[row_1, row_2, row_3]])
```
Comment (mf2199) [inline bigtable/tests/unit/test_table.py:1164]: Why do we need second array dimension here?

File: bigtable/google/cloud/bigtable/table.py (hunk 2 above)
Comment (pchauhan-qlogic) [reply bigtable/google/cloud/bigtable/table.py:51]: I have asked ajay for max thread limit and he replied with 10

File: bigtable/tests/unit/test_table.py (hunk 6 above)
Comment (pchauhan-qlogic) [reply bigtable/tests/unit/test_table.py:1164]: Now, we are passing batches not list of rows

File: bigtable/google/cloud/bigtable/table.py (hunk 5 above)
Comment (pchauhan-qlogic) [reply bigtable/google/cloud/bigtable/table.py:728]: added

File: bigtable/google/cloud/bigtable/table.py (hunk 4 above)
Comment (pchauhan-qlogic) [reply bigtable/google/cloud/bigtable/table.py]: added

Comment (mf2199) [reply bigtable/google/cloud/bigtable/table.py]: Why are we dropping the support for rows?

**Pull request closed without merging**